rolecall/rolecall.py

```diff
@@ -180,17 +218,89 @@ def __init__(self, bot):
             pass
 
         mcheck = lambda msg: msg.content.lower().startswith(('yes', 'no', 'cancel'))
-
         tasks = (self.bot.wait_for_message(author=author, timeout=15, channel=channel,
                                            check=mcheck),
                  self.bot.wait_for_reaction(user=author, timeout=15, message=message,
                                             emoji=('✅', '❌') ))
 
         converters = (lambda r: r.content.lower().startswith('yes'),
                       lambda r: r.reaction.emoji == '✅')
-
         return await wait_for_first_response(tasks, converters)
 
+    async def on_socket_raw_receive(self, msg):
+        """ Listens to reaction adds/removes and adds them to the reaction 
+        queue """
+
+        reaction = json.loads(msg)
+        if reaction['t'] == 'MESSAGE_REACTION_ADD' or reaction['t'] == 'MESSAGE_REACTION_REMOVE':
+            user_id = reaction['d']['user_id']
+            self.reaction_queue[user_id] = reaction
+            self.reaction_user_queue.add(user_id)
+
+    async def queue_processor(self):
+        """ Iterates the reaction queue every  0.0001 seconds. If reaction was 
+        added to a roleboard entry, corresponding role is assigned to user 
+        depending on the emoji pressed. If a reaction was removed, role is 
+        unassigned from user 
+        """
+
+        while True:
+            if self.reaction_user_queue:
+                next_key = self.reaction_user_queue.pop()
+                reaction = self.reaction_queue.pop(next_key)
+                await self.process_event(reaction)
+            await asyncio.sleep(0.0001)
```

Comment:
can probably wait longer than this 😬

Reply:
updated ridiculous sleep time to a less ridiculous 0.1 seconds as of commit 34af3cf